server/groups.js

```diff
@@ -53,56 +53,96 @@ module.exports = function(app, db, baseUrl) {
         });
     });
 
-    app.put(baseUrl + "/groups/:id", function(req, res) {
+    app.put(baseUrl + "/groups/:id/update", function(req, res) {
+        var userID = req.session.user;
         var groupID = req.params.id;
-        var groupInfo = req.body.groupInfo;
-        var newUsers = req.body.newUsers;
-        var removedUsers = req.body.removedUsers;
+        var groupInfo = req.body;
         var queryObject = {_id: new ObjectID(groupID)};
-        groups.find(queryObject).limit(1).next().then(function(group) {
-            if (!group) {
-                res.sendStatus(404);
-                return Promise.resolve();
-            }
-            if (group.users.indexOf(req.session.user) === -1) {
-                res.sendStatus(403);
-                return Promise.resolve();
+        findAndValidateGroup(userID, queryObject).then(function(group) {
+            var updateObject = {$set: {}};
+            for (var item in groupInfo) {
+                // Don't update any fields that don't exist, the _id field (locked), or the users field
+                // (limited access)
+                if (group[item] && item !== "_id" && item !== "users") {
```

Comment:
Right now, nothing; if the request body includes a field that is non-existent it's not included as part of the update document. I'm not sure if this is ideal behaviour or not, but it seemed simple and reasonable at the time.